server/controllers/user/profile.js

```diff
@@ -0,0 +1,33 @@
+const { Work, Review } = require('../../database/models');
+const paramsValidation = require('../../utils/validation/params');
+const { customError } = require('../errors');
+
+const profile = async (req, res, next) => {
+  try {
+    const { id } = await paramsValidation.validateAsync(req.params);
+    const works = await Work.findAll({
+      where: {
+        userId: id,
+      },
+    });
+    const user = await works[0].getUser();
+    const services = await user.getServices();
+    const locations = await user.getLocation();
```

Comment:
can you imagine how many await we have here,
Why not trying to fetch data using include to join tables we have
```js
await User.findByPk(id, {
      include: [{ model: Location }, { model: SubServices }],
      })
 ```
 So what if user here will be null ?! Do we need to send status 200, I doubt that
 So we need to check if user exists then maybe we can get reviews and works after that
 and while neither review nor work depends on each other we can run queries at the same time using `Promise.all`
 
 And did you notice that we're sending passwords to the client-side!! 😨 
 Also, it's good to not sending all raws even we don't use, So try to exclude columns we don't need